### PR TITLE
Add support for Berkshelf

### PR DESCRIPTION
**Pull request by andrewjamesbrown**
This adds preferential support for Berkshelf instead of Librarian for
uploading a chef-repo to a chef-zero server.  If a Berksfile is present,
it will be used, otherwise Librarian is used to upload dependent
cookbooks.  Note that 'berks upload' is not used, as that may upload to
the real chef server, though this is a future optimization to consider.

### CODE REVIEW

**Comment by dwradcliffe** [comment]
I think #1 should solve this problem without adding specific Berkshelf support

File: lib/chefdepartie.rb [hunk 1]
```diff
@@ -7,3 +7,5 @@
 require 'chefdepartie/cookbook'
 require 'chefdepartie/databag'
 require 'chefdepartie/cache'
+
+Chefdepartie.run(background: true, config: 'config.rb', cache: nil)
```
Comment (dalehamel) [inline lib/chefdepartie.rb]: was this done for testing? I don't think we want this hardcoded here

File: lib/chefdepartie/cookbook.rb [hunk 2]
```diff
@@ -28,6 +33,7 @@ def self.upload_cookbooks(path, books)
 
       books = books.collect do |name|
         next if Cache.cache(File.join(path, name))
+        next if File.file?(File.join(path, name))
```
Comment (dalehamel) [inline lib/chefdepartie/cookbook.rb:36]: I guess this is because it might erroneously try to upload a single file as a cookbook? Should we make this 'next unless File.dir?'?

File: lib/chefdepartie.rb (hunk 1 above)
Comment (andrewjamesbrown) [inline lib/chefdepartie.rb]: Yes it was done for testing.  Also was following the README, which doesn't work without this line...  i.e. `bundle exec ruby lib/chefdepartie.rb` simply returns without it.

File: lib/chefdepartie/cookbook.rb (hunk 2 above)
Comment (andrewjamesbrown) [inline lib/chefdepartie/cookbook.rb:36]: `next if File.file?` is the same as `next unless File.directory?`, isn't it?

**Comment by dalehamel** [comment]
> I think #1 should solve this problem without adding specific Berkshelf support

You're correct about that, I think that this change still probably has merit though, as most people either use librarian or berkshelf

**Comment by dalehamel** [comment]
If the tests were running in travis (which I should make them do), I think they would probably fail with this change, as they assert that it outputs 'uploading librarian cookboks'.

**Comment by andrewjamesbrown** [comment]
Fixed tests with 8d1890c

File: lib/chefdepartie/cookbook.rb (hunk 2 above)
Comment (dalehamel) [inline lib/chefdepartie/cookbook.rb:36]: I think there are conceivably edge cases, for special files (maybe for instance,  symlinks, where it might respond to File.file? with true, as well as File.directory? with true, if the target is a directory - unsure of this).

Comment (andrewjamesbrown) [inline lib/chefdepartie/cookbook.rb:36]: http://ruby-doc.org/core-2.2.0/File.html#method-c-file-3F

> Returns true if the named file exists and is a regular file.
> 
> file can be an IO object.
> 
> If the file argument is a symbolic link, it will resolve the symbolic link and use the file referenced by the link.

File: lib/chefdepartie.rb (hunk 1 above)
Comment (dalehamel) [inline lib/chefdepartie.rb]: good to merge once this is removed, I'll update the readme
